lib/sync.h

```diff
@@ -32,8 +32,9 @@ struct sync_cb {
 	int (*is_playing)(void *);
 };
 #define SYNC_DEFAULT_PORT 1338
-int sync_tcp_connect(struct sync_device *, const char *, unsigned short);
-int SYNC_DEPRECATED("use sync_tcp_connect instead") sync_connect(struct sync_device *, const char *, unsigned short);
+int SYNC_DEPRECATED("use sync_server_setup_tcp instead") sync_tcp_connect(struct sync_device *, const char *, unsigned short);
+int SYNC_DEPRECATED("use sync_server_setup_tcp instead") sync_connect(struct sync_device *, const char *, unsigned short);
+int sync_server_setup_tcp(struct sync_device *d, const char *host, unsigned short port);
```

Comment:
That could work.  I was more interested in the connection just transparently being maintained once setup via sync_update(), but since you seem to prefer connects being initiated explicitly this should probably be different.